### DISCUSSION TITLE
Branch Tagging / Grouping

### DISCUSSION THREAD

**Discussion by DLu** [community/community#61267] [Repositories]
### Select Topic Area

Product Feedback

### Body

It would be useful to have a way to way to group branches. 

 * In the branch dropdown on the main repository page, the default branch is shown first, followed by all the branches alphabetically. 
 * On the repo/branches page, the branches are divided by Default / Your branches / Active branches / Stale branches. 

There are [use cases](https://discourse.ros.org/t/random-hot-take-use-forks-for-core-repos/32422/5) where you want to have the branches be in the same repository and not a separate fork. However, right now there is no way to distinguish between the feature branches and core / trunk branches. 

 * In the branch dropdown, after the default branch, tagged branches could appear before all other alphabetic branches. 
 * You could filter branches on the branches page with the tags as well.

**Comment by spenserblack** [comment]
Personally, I name my branches like `feature/foo` or `bugfix/bar`, which helps (it also helps with auto-complete locally :wink:)

But branch protections and rulesets already identify branches by patterns, effectively grouping them. I wonder if those protections can be repurposed to also provide some grouping :thinking:
